native/common/jp_javaframe.cpp

```diff
@@ -1194,7 +1194,13 @@ jint JPJavaFrame::compareTo(jobject obj, jobject obj2)
 {
 	jvalue v;
 	v.l = obj2;
-	return CallIntMethodA(obj, m_Context->m_CompareToID, &v);
+	jint ret = m_Env->CallIntMethodA(obj, m_Context->m_CompareToID, &v);
+	if (m_Env->ExceptionOccurred())
+	{
+		m_Env->ExceptionClear();
```

Comment:
Sorry, I think I understand now. You're hesitating because the TypeError won't pop up at all on Pythons side. So instead we we could trigger a Python warning to inform the user about the badly behaving Java library, right?

Reply:
probably it is not worth the effort. This would have come up in the past already, if it would be really relevant. What do you think about emitting a warning, if the underlying lib raises an unexpected exception type?

Reply:
I will have to change some logic so that ClassCastException was available to compare to.   That isn't one that we have used previously so we don't have it available in the context.   Assuming that it is available I would need to call equals to check if it were the right type then call toString() to get the name of the Exception if it isn't then use C++ to formulate a custom warning string for the exception so that it could issue ``TypeError("Unexpected java exception type java.lang.RuntimeError encountered")``.  Does that sound correct?

Reply:
I think it does. But do you think it is worth your effort? I kinda feels like JPype is working around badly behaving Java libraries.

Reply:
I think that the chance that something throws a wrong exception and we give TypeError it would be fine.  Though perhaps something in the documentation would be needed so that one can call equals directly to get the actual.

Reply:
Having one example in docs showing this usage couldn't hurt, right?